### PR TITLE
Add org.hydrogenmusic.Hydrogen

### PR DESCRIPTION
**Pull request by hfiguiere**
I chose 1.0.0beta1 as the version because it uses qt5 which mean it can work with HiDPI screens.

### CODE REVIEW

File: org.hydrogenmusic.hydrogen.json [hunk 1]
```diff
@@ -0,0 +1,174 @@
+{
+  "app-id": "org.hydrogenmusic.hydrogen",
+  "branch": "stable",
```
Comment (bilelmoussaoui) [inline org.hydrogenmusic.hydrogen.json]: Not needed

File: org.hydrogenmusic.hydrogen.json [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.11",
```
Comment (bilelmoussaoui) [inline org.hydrogenmusic.hydrogen.json]: Have you tried 5.12?

File: org.hydrogenmusic.hydrogen.json [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  "sdk": "org.kde.Sdk",
+  "command": "hydrogen",
+  "rename-desktop-file": "hydrogen.desktop",
+  "rename-appdata-file": "hydrogen.appdata.xml",
```
Comment (bilelmoussaoui) [inline org.hydrogenmusic.hydrogen.json]: It would be nice if you could upstream those changes!

**Comment by bilelmoussaoui** [comment]
testbot, build

**Comment by bilelmoussaoui** [comment]
testbot, build org.hydrogenmusic.hydrogen

File: org.hydrogenmusic.hydrogen.json [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    "--socket=wayland",
+    /* Note playback */
+    "--socket=pulseaudio",
+    /* MIDI */
+    "--device=all",
+    /* Allow loading, saving files from anywhere (portals don’t work yet) */
+    "--filesystem=host",
+    /* Allow other instances to see lockfiles */
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_AUTO_SCREEN_SCALE_FACTOR=1"
```
Comment (barthalion) [inline org.hydrogenmusic.hydrogen.json]: This sounds odd. Is hydrogen UI broken somehow when scaled up?

File: org.hydrogenmusic.hydrogen.json [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+  ],
+  "modules": [
```
Comment (barthalion) [inline org.hydrogenmusic.hydrogen.json]: You should add global or per module cleanup section to remove development headers (and so on).

File: org.hydrogenmusic.hydrogen.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        {
+          "type": "archive",
+          "url": "http://download.sf.net/sourceforge/liblo/liblo-0.29.tar.gz",
+          "sha256": "ace1b4e234091425c150261d1ca7070cece48ee3c228a5612d048116d864c06a"
+        }
+      ]
+    },
+    {
+      "name": "hydrogen",
+      "buildsystem": "cmake",
```
Comment (barthalion) [inline org.hydrogenmusic.hydrogen.json]: cmake-ninja should be slightly faster

File: patches/hydrogen-cmake.patch [hunk 7]
```diff
@@ -0,0 +1,11 @@
+--- CMakeLists.txt	2018-03-18 07:50:43.000000000 -0400
```
Comment (barthalion) [inline patches/hydrogen-cmake.patch]: Has it been upstreamed?

**Comment by barthalion** [comment]
Have you contacted upstream team if they'd like to (co-)maintain the package?

File: org.hydrogenmusic.hydrogen.json (hunk 4 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: Not Hydrogen. Qt. Qt needs an opt-in to actually work properly on HiDPI. I have already submitted the patch upstream to make that env variable unnecessary.

File: patches/hydrogen-cmake.patch (hunk 7 above)
Comment (hfiguiere) [reply patches/hydrogen-cmake.patch]: not yet.

File: org.hydrogenmusic.hydrogen.json (hunk 2 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: jack1 fails to build on this 5.12.

Comment (bilelmoussaoui) [reply org.hydrogenmusic.hydrogen.json]: How fails exactly? Is it something that we can patch here or it's completely broken?

Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: it is missing db.h it seems.

File: org.hydrogenmusic.hydrogen.json (hunk 6 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: done

File: org.hydrogenmusic.hydrogen.json [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+      "config-opts": [
+        "-DWANT_JACK=0",
+        "-DWANT_PORTAUDIO=1",
+        "-DWANT_PORTMIDI=1"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://cfhcable.dl.sourceforge.net/project/hydrogen/Hydrogen/1.0.0%20Sources/hydrogen-1.0.0-beta1.tar.gz",
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: done

File: org.hydrogenmusic.hydrogen.json (hunk 5 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: done

File: patches/hydrogen-cmake.patch (hunk 7 above)
Comment (hfiguiere) [reply patches/hydrogen-cmake.patch]: A variant has been upstreamed hydrogen-music/hydrogen#689

File: org.hydrogenmusic.hydrogen.json [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+      "name": "hydrogen",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
+        "-DWANT_JACK=0",
+        "-DWANT_PORTAUDIO=1",
+        "-DWANT_PORTMIDI=1"
+      ],
+      "build-options": {
+        "env": [
+          "LADSPA_PATH=/app/lib/ladspa"
```
Comment (eszlari) [inline org.hydrogenmusic.hydrogen.json]: Don't you have to set this in `finish-args` too, to make the app find the files at runtime? At least, that's what i had to do for some apps I maintain.

File: org.hydrogenmusic.hydrogen.json [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+          "type": "archive",
+          "url": "http://download.sf.net/sourceforge/liblo/liblo-0.29.tar.gz",
+          "sha256": "ace1b4e234091425c150261d1ca7070cece48ee3c228a5612d048116d864c06a"
+        }
+      ]
+    },
+    {
+      "name": "hydrogen",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
```
Comment (eszlari) [inline org.hydrogenmusic.hydrogen.json]: `-DCMAKE_BUILD_TYPE=RelWithDebInfo`

File: org.hydrogenmusic.hydrogen.json (hunk 2 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/issues/608

**Comment by nedrichards** [comment]
testbot, build org.hydrogenmusic.hydrogen
bot, build org.hydrogenmusic.hydrogen

File: org.hydrogenmusic.hydrogen.json (hunk 3 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: see hydrogen-music/hydrogen#689

File: org.hydrogenmusic.hydrogen.json (hunk 2 above)
Comment (barthalion) [reply org.hydrogenmusic.hydrogen.json]: It would be better to just embed it in the application and switch to 5.12.

**Comment by barthalion** [comment]
Let's try to get it working on 5.12 instead of having to port it later.

**Comment by nedrichards** [comment]
testbot, build org.hydrogenmusic.hydrogen
bot, build org.hydrogenmusic.hydrogen

**Comment by nedrichards** [comment]
Build failed on aarch64 https://flathub.org/builds/#/builders/39/builds/697

**Comment by hfiguiere** [comment]
urgh. lash configure script is too old.

**Comment by nedrichards** [comment]
testbot, build org.hydrogenmusic.hydrogen
bot, build org.hydrogenmusic.hydrogen

File: org.hydrogenmusic.hydrogen.json (hunk 2 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: Apparently it wasn't meant to be removed. And libdb doesn't seem to be easy (a tonne of patches amongst other things)

Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: WiP to be fixed: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/merge_requests/929

**Comment by eszlari** [comment]
Are you sure about the `app-id`? Should'nt the name be capitalized: `org.hydrogenmusic.Hydrogen`

Maybe you should contact upstream about it.

File: org.hydrogenmusic.hydrogen.json (hunk 2 above)
Comment (hfiguiere) [reply org.hydrogenmusic.hydrogen.json]: Bumped jack to 1.9.12 and lash seems to be api with it. SDK 5.12 now.

**Comment by hfiguiere** [comment]
Upstream didn't have an app-id. I change it now.

**Comment by nedrichards** [comment]
testbot, build org.hydrogenmusic.Hydrogen
bot, build org.hydrogenmusic.Hydrogen

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.hydrogenmusic.Hydrogen

**Pull request closed without merging**